packages/server/src/plugin/cacheControl/index.ts

```diff
@@ -87,20 +61,54 @@ export function ApolloServerPluginCacheControl(
       // versions of Gateway older than 0.35.0, we should also run this code
       // from a schemaDidLoadOrUpdate instead of serverWillStart. Using
       // schemaDidLoadOrUpdate throws when combined with old gateways.)
-      typeAnnotationCache.max = Object.values(schema.getTypeMap()).filter(
-        isCompositeType,
-      ).length;
-      fieldAnnotationCache.max =
-        Object.values(schema.getTypeMap())
-          .filter(isObjectType)
-          .flatMap((t) => Object.values(t.getFields())).length +
-        Object.values(schema.getTypeMap())
-          .filter(isInterfaceType)
-          .flatMap((t) => Object.values(t.getFields())).length;
+      typeAnnotationCache = new LRUCache<GraphQLCompositeType, CacheAnnotation>(
```

Comment:
This is a case where it's a bit of a bummer that we didn't make requestDidStart a method (optionally?) returned from the (introduced later) serverWillStart rather than top-level, since then you could just have a `const` here.